### PR TITLE
WIP (PUP-3899) Add tests for file checksum

### PR DESCRIPTION
**Pull request by MikaelSmith**
Add integration and acceptance tests for specifying the checksum on a file. Covers puppet apply, querying the server via the HTTP API, and puppet agent runs.

### CODE REVIEW

File: acceptance/tests/resource/file/content_attribute.rb [hunk 1]
```diff
@@ -15,6 +15,27 @@
...
+  end
+
+  ['mtime', 'ctime'].each do |checksum_type|
+    step "Content Attribute: using #{checksum_type}"
+    manifest = "file { '#{target}': content => 'This is the test file content', ensure => present, checksum => #{checksum_type} }"
+    apply_manifest_on agent, manifest, :acceptable_exit_codes => [1] do
+      assert_match(/Error: Validation of File\[#{target}\] failed: You cannot specify content when using checksum '#{checksum_type}'/, stderr, "#{agent}: expected failure")
+    end
+  end
+
```
Comment (joshcooper) [inline acceptance/tests/resource/file/content_attribute.rb]: Could you write this as a single manifest with multiple file resources each with a unique name? It turns out to be much faster than running `puppet agent` a bunch

Comment (MikaelSmith) [inline acceptance/tests/resource/file/content_attribute.rb]: Oh yeah, sure. I could do that for acceptance tests too.

**Comment by joshcooper** [comment]
Looks like a merge conflict on this one, likely due to change for PUP-3898

**Comment by er0ck** [comment]
i'm :+1: after josh's requested changes above

**Comment by MikaelSmith** [comment]
Creating a new PR because I had to make a bunch of changes.

**Pull request closed without merging**